src/main/groovy/io/gatling/gradle/GatlingRunTask.groovy

```diff
@@ -75,7 +75,12 @@ class GatlingRunTask extends DefaultTask {
     void gatlingRun() {
         def gatlingExt = project.extensions.getByType(GatlingPluginExtension)
 
-        Map<String, ExecResult> results = simulationFilesToFQN().collectEntries { String simulationClass ->
+        def resolvedSimulations = simulationFilesToFQN()
+        if (resolvedSimulations.isEmpty()) {
+            throw new IllegalArgumentException("No simulation matching 'simulations' closure found")
```

Comment:
Not sure if this is a wanted behavior.
See my comment about aggregated tasks https://github.com/gatling/gatling/issues/4042#issuecomment-1367937056

Note: there are tests to change

Reply:
If the project has sub-projects, wouldn't the recommended solution be `gradle -p subproject gatlingRun`? Same thing for sbt: I always felt super awkward to run `testOnly` from root and waste time scanning all the modules.